tests/FormlyMapper/FormlyField/BirthdayFieldTest.php

```diff
@@ -0,0 +1,280 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Linio\DynamicFormBundle\Tests\FormlyMapper\FormlyField;
+
+use Linio\DynamicFormBundle\Exception\InvalidConfigurationException;
+use Linio\DynamicFormBundle\Exception\NumberFormatException;
+use Linio\DynamicFormBundle\FormlyMapper\FormlyField\BirthdayField;
+use PHPUnit\Framework\TestCase;
+
+class BirthdayFieldTest extends TestCase
+{
+    /**
+     * @var DateField
+     */
+    protected $formlyField;
+
+    public function basicDataProvider(): array
+    {
+        $config = [
+            'name' => 'borndate',
+            'type' => 'birthday',
+            'options' => [
+                'required' => true,
+                'label' => 'Borndate',
+            ],
+        ];
+
+        $expected = [
+            'key' => 'borndate',
+            'type' => 'input',
+            'templateOptions' => [
+                'required' => true,
+                'label' => 'Borndate',
+                'type' => 'birthday',
+            ],
+        ];
+
+        return [
+            [$config, $expected],
+        ];
+    }
+
+    /**
+     * @dataProvider basicDataProvider
+     */
+    public function testIsAddingBirthdayFields(array $fieldConfiguration, array $expected): void
+    {
+        $fieldConfiguration['validation'] = [
+            'Symfony\Component\Validator\Constraints\Regex' => [
+                'pattern' => '^[0-9]{4}\-[0-9]{2}\-[0-9]{2}$',
+                'message' => 'The birthday field must follow the pattern "yyyy-MM-dd"',
+            ],
+        ];
+
+        $expected['templateOptions']['pattern'] = '^[0-9]{4}\-[0-9]{2}\-[0-9]{2}$';
+        $expected['validation'] = [
+            'messages' => [
+                'regex' => 'The birthday field must follow the pattern "yyyy-MM-dd"',
+            ],
+        ];
+
+        $this->formlyField->setFieldConfiguration($fieldConfiguration);
+        $actual = $this->formlyField->getFormlyFieldConfiguration();
+
+        $this->assertEquals($expected, $actual);
+    }
+
+    /**
+     * @dataProvider basicDataProvider
+     */
+    public function testIsAddingSpecificYears(array $fieldConfiguration, array $expected): void
+    {
+        $years = [2000, 2001, 2002, 2003, 2004, 2005];
+
+        $fieldConfiguration['options']['years'] = $years;
+
+        $expected['templateOptions']['years'] = $years;
+
+        $this->formlyField->setFieldConfiguration($fieldConfiguration);
+        $actual = $this->formlyField->getFormlyFieldConfiguration();
+
+        $this->assertEquals($expected, $actual);
+    }
+
+    /**
+     * @dataProvider basicDataProvider
+     */
+    public function testIsThrowingExceptionWhenYearsArrayHasNonNumericalValues(array $fieldConfiguration, array $expected): void
```

Comment:
Is the business use case though that it throws an exception, or is the business use case that it does not all non-numerical values? Test names should be written with the business case in mind, not the technical implementation.

In essence, throwing the exception is HOW you make sure that it does not accept non-numerical values, but the test is only concerned with the what, which is that it does not accept them.

The easy way to tell if a test is too specific is usually by checking the name of the test to see if it references programming, or if it reads more like a story from a product manager. Test names are there to tell a story about what you are trying to verify.

Reply:
Excellent feedback! please don't stop commenting :) I am learning a lot thanks to you.
I will review all the names of the tests.

Reply:
No problem at all. I'm also aware that English is not your first language, so no worries there. I perform all my reviews with that fact in mind since you don't always know who the other person is, their circumstances, etc.